src/include/buffer.h

```diff
@@ -304,7 +310,6 @@ namespace buffer CEPH_BUFFER_API {
 
       void advance(int o);
       void seek(unsigned o);
-      bool operator!=(const iterator_impl& rhs) const;
```

Comment:
I don't think we should bump the so version in this case, since it will cause issues with librados C users (e.g. packages like qemu or libvirt that depend on librados2), and it would only affect a c++ librados program compiled against infernalis iiuc. The move to c++11 in infernalis required a recompile for c++ librados users already, so another one for jewel doesn't seem worth the trouble an so version bump brings (especially this close to jewel).